workitem/enum_type_blackbox_test.go

```diff
@@ -321,3 +321,86 @@ func TestEnumType_EqualEnclosing(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumType_ConvertFromModel(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	type testCase struct {
+		subTestName    string
+		input          interface{} // contains valid and invalid values
+		expectedOutput interface{}
+		wantErr        bool
+	}
+	tests := []struct {
```

Comment:
The enum type created for the test uses only `string, int and float` as the base type. I have not added `area/iteration/user` for base type because it would not add much value.

Reply:
It **would** add much value but currently the type-system treats all of them as strings which is simple wrong. But this is subject of another PR.